src/Service/ActivityPub/ApHttpClient.php

```diff
@@ -529,16 +529,26 @@ private function getHeaders(string $url, User|Magazine $actor, ?array $body = nu
         $stringToSign = self::headersToSigningString($headers);
         $signedHeaders = implode(' ', array_map('strtolower', array_keys($headers)));
         $key = openssl_pkey_get_private($actor->privateKey);
-        openssl_sign($stringToSign, $signature, $key, OPENSSL_ALGO_SHA256);
-        $signature = base64_encode($signature);
-
-        $keyId = $actor instanceof User
-            ? $this->personFactory->getActivityPubId($actor).'#main-key'
-            : $this->groupFactory->getActivityPubId($actor).'#main-key';
-
-        $signatureHeader = 'keyId="'.$keyId.'",headers="'.$signedHeaders.'",algorithm="rsa-sha256",signature="'.$signature.'"';
+        $success_sign = openssl_sign($stringToSign, $signature, $key, OPENSSL_ALGO_SHA256);
+        // Free the key from memory
+        openssl_free_key($key);
+        $signatureHeader = null;
+        if ($success_sign) {
+            $signature = base64_encode($signature);
+            $keyId = $actor instanceof User
+                ? $this->personFactory->getActivityPubId($actor).'#main-key'
+                : $this->groupFactory->getActivityPubId($actor).'#main-key';
+            $signatureHeader = 'keyId="'.$keyId.'",headers="'.$signedHeaders.'",algorithm="rsa-sha256",signature="'.$signature.'"';
+        } else {
+            $this->logger->error('Failed to sign headers for {url}: {headers}', [
+                'url' => $url,
+                'headers' => $headers,
+            ]);
```

Comment:
Should you return here (and at 2nd comment) or throw an exception so this won't get processed any further?

Reply:
Yea so what's the big question as stated in my pr description. What should we do? 🤔

It's ssl signing that important to fail or not? What are the consequences of both solutions?

Reply:
This methods should never fail. Maybe add an additional check at the top of the method whether the actor is a local one. If the actor is local then there is literally no reason for not having a private key...

Reply:
OK. I will just throw an error.